homeassistant/components/sensor/__init__.py

```diff
@@ -179,13 +181,17 @@ def add_to_platform_start(
     ) -> None:
         """Start adding an entity to a platform.
 
+        Raise HomeAssistantError if sensor has the entity category config as a sensor cannot change something.
+
         Allows integrations to remove legacy custom unit conversion which is no longer
         needed without breaking existing sensors. Only works for sensors which are in
         the entity registry.
-
-        This can be removed once core integrations have dropped unneeded custom unit
-        conversion.
         """
+        if self.entity_category == EntityCategory.CONFIG:
+            raise HomeAssistantError(
+                f"Entity {type(self)} cannot be added as the entity category is set to config"
+            )
+
         super().add_to_platform_start(hass, platform, parallel_updates)
```

Comment:
Added the code before this line, so `_platform_state` will still be `NOT_ADDED`. It will be changed in the parent function to `ADDED`